### DISCUSSION TITLE
FTMS not working with Echelon Stride and Peloton App anymore?

### DISCUSSION THREAD

**Discussion by speedf1** [cagnulein/qdomyos-zwift#2597] [Q&A] [answered]
It's probably something I've done, but I'm at a loss as to what has happened.  

I have two profiles set on the QA app.  One for the Echelon Sport bike and one for the Echelon Stride treadmill.  Stats for both devices are picked up by the QZ app and displayed within the app.  However, only the bike stats are picked up by the IOS Peloton app.  The Peloton app doesn't seem to recognise the stats transmitted by the QZ app for the stride.  

This did use to work - I'm not aware of my changing anything though.

Attached are debug logs

[debug-Thu_Sep_12_16_22_51_2024.log](https://github.com/user-attachments/files/17001498/debug-Thu_Sep_12_16_22_51_2024.log)

**Comment by cagnulein** [comment]
Since I didn't change anything I guess peloton updates their app? I will
test this when I will be at home

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 14 set 2024 alle 09:53 speedf1 ***@***.***> ha
scritto:

**Reply by speedf1** [reply]
Hi Roberto, 

Once again, I'm impressed by how quickly you respond - are you just sitting awaiting these messages coming in?  ;-)

Take your time, I wouldn't be surprised if it's something that I've done!

Thanks again!

**Comment by cagnulein** [comment]
Nope actually I'm at the park with my little daughter  :)

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 14 set 2024 alle 10:01 speedf1 ***@***.***> ha
scritto:

**Reply by cagnulein** [reply]
This works

![Screenshot_20240914-123025.png](https://github.com/user-attachments/assets/b37a74d9-6b75-42a3-95ba-ba70cf97f524)

Let me try the last one

**Reply by cagnulein** [reply]
also the last one works

![image](https://github.com/user-attachments/assets/ebad32b3-525a-496a-aec2-3ef2de80b068)

**Reply by cagnulein** [reply]
![Screenshot_20240914-124046.png](https://github.com/user-attachments/assets/73ced722-8cdf-4f05-9e5c-d86ab8000373)

@speedf1 install nrfconnect on the peloton app and show me how you see qz from there

**Comment by speedf1** [comment]
Am away for a couple of days, will do that when I get back.

**Reply by cagnulein** [reply]
Sure

**Comment by speedf1** [comment, marked as answer]
OK, I think I've worked out the problem.

It's something to do with the Bluetooth name of the device running the QZ app.

If I have the bluetooth name set to "Ians OnePlus 10T", then the iPad running the Peloton app can't "see" the QZ stats.  If I change the bluetooth name to "Ians OnePlus" (i.e. just reduce the name by a few characters), then the Peloton app CAN see the QZ stats.

Interestingly this is only a problem on the QZ profile I've created for the treadmill and not for the bike.

**Reply by cagnulein** [reply]
yes it could be because for treadmill qz is sending more bytes in the advertise channel. so less byte for the names are leaving more bytes for other data.

**Comment by speedf1** [comment]
That certainly sounds reasonable.  Is it worth adding a check to the QZ app and then advising users if the bluetooth name is too long?

**Reply by cagnulein** [reply]
Yes thanks for the idea! I will check if it's doable!
